azure-pipelines.yml

```diff
@@ -121,6 +130,91 @@ jobs:
         ADO_PULL_REQUEST_LATEST_COMMIT: HEAD
         ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
 
+- job: getPythonVersion
+  displayName: "get Python Version"
+  pool:
+    name: 'pool-ubuntu-2004'
+  steps:
+    - task: UsePythonVersion@0
+      displayName: 'Use Python 3.10'
+      inputs:
+        versionSpec: '3.10'
+    - template: .azure-pipelines/templates/azdev_setup.yml
+    - bash: pip install wheel==0.30.0
+      displayName: 'Install wheel==0.30.0'
+    - bash: |
+        #!/usr/bin/env bash
+        set -ev
+
+        source env/bin/activate
+        az --version
+        python scripts/ci/get_python_version.py
+        pythonVersion=`cat python_version.txt`
+        echo $pythonVersion
+        echo "##vso[task.setvariable variable=pythonVersion;isoutput=true]$pythonVersion"
+      name: CustomParas
+      displayName: 'get Python Version'
+      env:
+        ADO_PULL_REQUEST_LATEST_COMMIT: HEAD
+        ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+
+- ${{each version in parameters.versions}}:
+  - job: MinVersionTests${{version.key}}
+    dependsOn: getPythonVersion
+    variables:
+      PythonVersion: $[ dependencies.getPythonVersion.outputs['CustomParas.pythonVersion'] ]
+    condition: and(succeeded(), contains(variables.PythonVersion, ${{version.value}}))
+    displayName: "Min Version Tests ${{version.key}}"
+    pool:
+      name: 'pool-ubuntu-2004'
+    steps:
+      - task: UsePythonVersion@0
+        displayName: 'Use ${{version.key}}'
+        inputs:
+          versionSpec: '${{version.value}}'
+          addToPath: true
+      - template: .azure-pipelines/templates/azdev_setup.yml
+      - bash: pip install wheel==0.30.0
+        displayName: 'Install wheel==0.30.0'
+      - bash: |
+          set -ev
+
+          . env/bin/activate
+          az --version
+          python scripts/ci/test_min_version.py -v
+        displayName: 'Run min version test ${{version.key}}'
+        env:
+          ADO_PULL_REQUEST_LATEST_COMMIT: HEAD
+          ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+
+- job: MinVersionTestsPython310
```

Comment:
DevOps pipeline convert python version 3.10 to 3.1.
So we need a separate job for python 3.10.
I've created an issue here：https://github.com/microsoft/azure-pipelines-tasks/issues/17323